### PR TITLE
Comment delete single

### PR DESCRIPTION
**Pull request by aroquev00**
This PR is for the added "Delete single comment" button. The button triggers a POST request to the DeleteServlet.java that "calculates" the key of the entity of the comment and then tells the DS to delete it.

image

Closes https://github.com/aroquev00/step191-2020/issues/12

### CODE REVIEW

File: aroquev/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import java.io.BufferedReader;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that updates upvotes or downvotes of a comment*/
```
Comment (brettapeters) [inline aroquev/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java:31]: Update doc comment

File: aroquev/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+@WebServlet("/delete-comment")
+public class DeleteCommentServlet extends HttpServlet { 
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String jsonString = deserializeJson(request);
```
Comment (brettapeters) [inline aroquev/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java:37]: Same comment as your previous PR about not needing the intermediate JSON string.

File: aroquev/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    String line = null;
+    try {
+        BufferedReader reader = request.getReader();
+        while ((line = reader.readLine()) != null)
+            jsonBuff.append(line);
+    } catch (Exception e) { 
+      /*error*/ 
+    }
+
+    return jsonBuff.toString();
```
Comment (seanjohnite) [inline aroquev/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java]: You can also use the Java Streams API to do this a little more cleanly:

```suggestion
    return request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
```

https://stackoverflow.com/questions/8100634/get-the-post-request-body-from-httpservletrequest

File: aroquev/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java (hunk 2 above)
Comment (aroquev00) [reply aroquev/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java]: True I forgot to change the other usage. Done.

File: aroquev/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java (hunk 1 above)
Comment (aroquev00) [reply aroquev/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java]: Done, changed all the other wrong comments as well.

File: aroquev/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java (hunk 3 above)
Comment (aroquev00) [reply aroquev/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java]: Ohhh coool, thanks!

**Pull request merged**